CONTRIBUTING.md

```diff
@@ -1,81 +1,120 @@
+### Build
 
 ```bash
 cargo build --target wasm32-unknown-unknown
 ```
 
-#### Examples
+#### stdweb
 
-TODO: Add more info 
-
-##### stdweb
-In order to run the examples in `./yew-stdweb`, you may wish to install [cargo-web]:
+In order to run the examples in `./yew-stdweb`, you may wish to install [cargo-web](https://github.com/koute/cargo-web):
 
 ```bash
 cargo install cargo-web
 ```
 
-[cargo-web]: https://github.com/koute/cargo-web
+## Tests
 
-#### Test
+Yew has unit tests which can be run with `cargo test`. However, a large portion of the tests need to be run in a browser.
+This section will guide you through the process of running these tests locally.
 
-##### Web Tests
-First, ensure that `wasm-bindgen-cli` is installed.
-[Instructions](https://rustwasm.github.io/docs/wasm-bindgen/wasm-bindgen-test/usage.html#install-the-test-runner)
+### Integration Tests
 
-Additionally a webdriver must be installed locally and configured to be on the
-`PATH`. Currently supports `geckodriver`, `chromedriver`, and `safaridriver`,
-although more driver support may be added! You can download these at:
+First, ensure that `wasm-pack` is installed.
+[Instructions](https://rustwasm.github.io/wasm-pack/installer/)
 
-* geckodriver - https://github.com/mozilla/geckodriver/releases
-* chromedriver - http://chromedriver.chromium.org/downloads
-* safaridriver - should be preinstalled on OSX
+`wasm-pack` automatically takes care of installing the correct `wasm-bindgen` version and downloading a webdriver for the browser.
 
-Lastly, the fetch tests require a local httpbin server running on port `8000`. We recommend running this with Docker:
+The following command is all you need to run the tests:
 
 ```bash
-docker run -p 8000:80 kennethreitz/httpbin
+wasm-pack test --firefox --headless -- --features wasm_test
 ```
 
-##### Macro Tests
-When adding or updating tests, please make sure you have updated the appropriate `stderr` file, which you can find [here](https://github.com/yewstack/yew/tree/master/yew-macro/tests/macro) for the `html!` macro. These files ensure that macro compilation errors are correct and easy to understand.
+You can replace `--firefox` with `--chrome` or `--safari` if you want to run the tests in a different browser. Currently Yew's CI tests use Firefox.
 
-To update or generate a new `stderr` file you can run `TRYBUILD=overwrite cargo test --test macro_test` or `TRYBUILD=overwrite cargo test --test derive_props_test` from the `yew-macro` directory.
+Make sure to run the command in the directory of the crate you wish to test.
+If you run the command in the repository root you will get an error like this:
+
+```bash
+Error: failed to parse manifest: yew/Cargo.toml
+Caused by: missing field `package`
+```
+
+#### Manually installing a webdriver
+
+`wasm-pack` automatically downloads the selected webdriver for you.
+If desired, you can download it manually. For instructions, please consult the documentation of the webdriver you wish to install.
```

Comment:
```suggestion
If desired, you can download a WebDriver manually. For instructions, please consult the documentation of the webdriver you wish to install.
```

Reply:
I'm going to adjust the other occurrences of "webdriver" to this capitalisation too.